homeassistant/components/sensor/syncthru.py

```diff
@@ -0,0 +1,244 @@
+    for key in printer.drumStatus(filter_supported=True):
+        if 'drum_' + str(key) in monitored:
+            devices.append(SyncThruDrum(hass, printer, name, key))
+    for key in printer.inputTrayStatus(filter_supported=True):
+        if 'tray_' + str(key) in monitored:
+            devices.append(SyncThruInputTray(hass, printer, name, key))
+    for key in printer.outputTrayStatus():
+        if 'output_tray_' + str(key) in monitored:
+            devices.append(SyncThruOutputTray(hass, printer, name, key))
+
+    add_devices(devices, True)
+    return True
+
+
+class SyncThruSensor(Entity):
+    """Implementation of an abstract Samsung Printer sensor platform."""
+
+    def __init__(self, hass, syncthru, name):
+        """Initialize the sensor."""
+        self._hass = hass
+        self.syncthru = syncthru
+        self._attributes = {}
+        self._state = STATE_UNKNOWN
+        self._name = name
+        self._friendly_name = name
+        self._icon = 'mdi:printer'
+        self._unit_of_measurement = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def icon(self):
+        """Return the icon of the device."""
+        return self._icon
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measuremnt."""
+        return self._unit_of_measurement
+
+    @property
+    def device_state_attributes(self):
+        """Return the state attributes of the device."""
+        self._attributes[CONF_FRIENDLY_NAME] = self._friendly_name
+        return self._attributes
+
+
+class SyncThruMain(SyncThruSensor):
+    """Implementaion of the main sensor, monitoring the general state."""
+
+    def update(self):
+        """Get the latest data from SyncThru and update the state."""
+        self.syncthru.update()
+        self._state = self.syncthru.deviceStatus()
+
+        if self.syncthru.isOnline():
+            self._friendly_name = self.syncthru.model()
+            self._state = self.syncthru.deviceStatus()
+
+
+class SyncThruToner(SyncThruSensor):
+    """Implementation of a Samsung Printer toner sensor platform."""
+
+    def __init__(self, hass, syncthru, name, color):
+        """Initialize the sensor."""
+        super().__init__(hass, syncthru, name)
+        self._name = "{} toner {}".format(name, color)
+        self._color = color
+        self._unit_of_measurement = '%'
+
+    def update(self):
+        """Get the latest data from SyncThru and update the state."""
+        # Data fetching is taken care of through the Main sensor
+
+        if self.syncthru.isOnline():
+            self._attributes = self.syncthru.tonerStatus(
+                ).get(self._color, {})
+            self._state = self._attributes.get(
+                'remaining', STATE_UNKNOWN)
```

Comment:
Don't use `STATE_UNKNOWN`. Use `None`, which is the default value returned if key is missing in the dict.